nixos/modules/services/mail/listmonk/default.nix

```
@@ -0,0 +1,181 @@
+        description = lib.mdDoc ''Database settings will be reset to the value set in this module if this is not enabled.
+        Enable this if you want to persist changes you have done in the application.
+        '';
+      };
+      databaseSettings = mkOption {
+        default = null;
+        type = with types; nullOr (submodule {
+          freeformType = oneOf [ (listOf str) (listOf (attrsOf anything)) str int bool ];
+
+          options = {
+            "app.notify_emails" = mkOption {
+              type = listOf str;
+              default = [];
+              description = lib.mdDoc "Administrator emails for system notifications";
+            };
+
+            "privacy.exportable" = mkOption {
+              type = listOf str;
+              default = ["profile" "subscriptions" "campaign_views" "link_clicks"];
+              description = lib.mdDoc "List of fields which can be exported through an automatic export request";
+            };
+
+            "privacy.domain_blocklist" = mkOption {
+              type = listOf str;
+              default = [];
+              description = lib.mdDoc "E-mail addresses with these domains are disallowed from subscribing.";
+            };
+
+            smtp = mkOption {
+              type = listOf (submodule (import ./smtp.nix));
+              description = lib.mdDoc "List of outgoing SMTP servers";
+            };
+
+            # TODO: refine this type based on the smtp one.
+            "bounce.mailboxes" = mkOption {
+              type = listOf (submodule {
+                freeformType = with types; oneOf [ str int bool ];
+              });
+              default = [];
+              description = lib.mdDoc "List of bounce mailboxes";
+            };
+
+            messengers = mkOption {
+              type = listOf str;
+              default = [];
+              description = lib.mdDoc "List of messengers, see: <https://github.com/knadh/listmonk/blob/master/models/settings.go#L64-L74> for options.";
+            };
+          };
+        });
+
+        description = lib.mdDoc "Dynamic settings in the PostgreSQL database, set by a SQL script, see <https://github.com/knadh/listmonk/blob/master/schema.sql#L177-L230> for details.";
+      };
+      secretFile = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = lib.mdDoc "Secret file loaded as an environment file, read <https://listmonk.app/docs/configuration/#environment-variables> for details.";
+      };
+    };
+  };
+
+
+  ###### implementation
+  config = mkIf cfg.enable (mkMerge [
+    {
+      services.listmonk.settings."app".address = mkDefault "localhost:9000";
+
+      # Default parameters from https://github.com/knadh/listmonk/blob/master/config.toml.sample
+      services.listmonk.settings."db".max_open = mkDefault 25;
+      services.listmonk.settings."db".max_idle = mkDefault 25;
+      services.listmonk.settings."db".max_lifetime = mkDefault "300s";
```

```suggestion
      # Default parameters from https://github.com/knadh/listmonk/blob/master/config.toml.sample
      services.listmonk.settings = {
        "db".max_open = mkDefault 25;
        "db".max_idle = mkDefault 25;
        "db".max_lifetime = mkDefault "300s";
      };
```

You in case you did not know: that works :)

---

Yeah, I knew, I just forgot do it, thanks!